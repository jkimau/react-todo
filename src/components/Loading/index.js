import React from 'react';
import styled from 'styled-components';

const LoadingDiv = styled.div`
  position: absolute;
  top: 50%;
  left: 50%;
  margin-top: -9px;
  margin-left: -35px;
`;

const Loading = () => (
  <LoadingDiv>Loading...</LoadingDiv>
);

export default Loading;
