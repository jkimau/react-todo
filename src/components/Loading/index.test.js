import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Loading from '../Loading';

describe('<Loading/> component', () => {
  const compileComponent = () => {
    return shallow(<Loading/>);
  }

  test('snapshot', () => {
    const tree = renderer.create(
      <Loading/>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should render Loading... text', () => {
    const compiledComponent = compileComponent();

    expect(compiledComponent.dive().find('div').text()).toEqual('Loading...');
  });
});
