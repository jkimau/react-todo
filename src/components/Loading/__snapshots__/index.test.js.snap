exports[`<Loading/> component snapshot 1`] = `
<div
  className="khmhuD">
  Loading...
</div>
`;
